src/app/pages/home/home.page.ts

```diff
@@ -1,19 +1,50 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { Router, NavigationExtras, ActivatedRoute } from '@angular/router';
 
+import { AuthService } from 'src/app/services/auth/auth.service';
+import { UserService } from 'src/app/services/user/user.service';
 
 @Component({
   selector: 'app-home',
   templateUrl: 'home.page.html',
   styleUrls: ['home.page.scss'],
 })
-export class HomePage {
+export class HomePage implements OnInit{
 
-  constructor(private route: ActivatedRoute, private router: Router) {
+  constructor(private route: ActivatedRoute, private router: Router, private authService: AuthService, private userService: UserService) {
     console.log("Home page started (constructor)");
   }
 
   ngOnInit() {
-    console.log("Home page started (init)")
+    console.log("Home page started (init)");
+    // check for a user entry by the uid
+    let goToSuccessfulSignupPageIfNoData =
+
+    // get the current user's uid
+    this.authService.getCurrentUser((error, currentUser) => {
+      if (!currentUser) {
+        console.log('Home Page : Currently logged out');
+        return;
+      }
+      this.userService.getUser(currentUser.uid,  (err, data) => {
+        if (err) {
+          console.log(err);
+        }
+        console.log(data);
+        if (!data) {
+          // go to Successful Signup Page if no user exists
+          let navigationExtras: NavigationExtras = {
+                  state: {
+                    uid: currentUser.uid,
+                    firstName: '',
+                    lastName: '',
+                    email: '',
+                    mobile: ''
+                  }
+                };
+          this.router.navigateByUrl('/successful-signup', navigationExtras);
```

Comment:
Yes I would prefer not to nest these calls but it's not so bad in this case. If there were more extensive branching logic then it would be harder to read. Will probably have to break it up for testability.